src/sas/qtgui/Perspectives/Fitting/UnitTesting/FittingWidgetTest.py

```diff
@@ -128,381 +124,384 @@ def customModels(cls):
+        assert mag_index.data() == "up_frac_f"
         mag_index = fittingWindow.lstMagnetic.model().index(2,0)
-        self.assertEqual(mag_index.data(), "up_theta")
+        assert mag_index.data() == "up_theta"
         mag_index = fittingWindow.lstMagnetic.model().index(3,0)
-        self.assertEqual(mag_index.data(), "up_phi")
+        assert mag_index.data() == "up_phi"
         mag_index = fittingWindow.lstMagnetic.model().index(4,0)
-        self.assertEqual(mag_index.data(), "sld_M0")
+        assert mag_index.data() == "sld_M0"
         mag_index = fittingWindow.lstMagnetic.model().index(5,0)
-        self.assertEqual(mag_index.data(), "sld_mtheta")
+        assert mag_index.data() == "sld_mtheta"
         mag_index = fittingWindow.lstMagnetic.model().index(6,0)
-        self.assertEqual(mag_index.data(), "sld_mphi")
+        assert mag_index.data() == "sld_mphi"
         mag_index = fittingWindow.lstMagnetic.model().index(7,0)
-        self.assertEqual(mag_index.data(), "sld_solvent_M0")
+        assert mag_index.data() == "sld_solvent_M0"
         mag_index = fittingWindow.lstMagnetic.model().index(8,0)
-        self.assertEqual(mag_index.data(), "sld_solvent_mtheta")
+        assert mag_index.data() == "sld_solvent_mtheta"
         mag_index = fittingWindow.lstMagnetic.model().index(9,0)
-        self.assertEqual(mag_index.data(), "sld_solvent_mphi")
+        assert mag_index.data() == "sld_solvent_mphi"
 
         # test the delegate a bit
         delegate = fittingWindow.lstMagnetic.itemDelegate()
-        self.assertEqual(delegate.editableParameters(), [1, 2, 3])
+        assert delegate.editableParameters() == [1, 2, 3]
 
-    def testSelectStructureFactor(self):
+    def testSelectStructureFactor(self, widget):
         """
         Test if structure factors have been loaded properly, including plugins also classified as a structure factor.
         """
-        fittingWindow =  self.widget
+        fittingWindow =  widget
 
         #Test for existence in combobox
-        self.assertNotEqual(fittingWindow.cbStructureFactor.findText("stickyhardsphere"),-1)
-        self.assertNotEqual(fittingWindow.cbStructureFactor.findText("hayter_msa"),-1)
-        self.assertNotEqual(fittingWindow.cbStructureFactor.findText("squarewell"),-1)
-        self.assertNotEqual(fittingWindow.cbStructureFactor.findText("hardsphere"),-1)
-        self.assertNotEqual(fittingWindow.cbStructureFactor.findText("plugin_structure_template"),-1)
-        self.assertEqual(fittingWindow.cbStructureFactor.findText("plugin_template"),-1)
+        assert fittingWindow.cbStructureFactor.findText("stickyhardsphere") != -1
+        assert fittingWindow.cbStructureFactor.findText("hayter_msa") != -1
+        assert fittingWindow.cbStructureFactor.findText("squarewell") != -1
+        assert fittingWindow.cbStructureFactor.findText("hardsphere") != -1
+        assert fittingWindow.cbStructureFactor.findText("plugin_structure_template") != -1
```

Comment:
All tests that aren't skipped pass on my machine (Win10), except this one. The error I'm seeing:
```
assert fittingWindow.cbStructureFactor.findText("plugin_structure_template") != -1
E       AssertionError: assert -1 != -1
E        +  where -1 = <built-in method findText of QComboBox object at 0x00000296F6FB33A0>('plugin_structure_template')
E        +    where <built-in method findText of QComboBox object at 0x00000296F6FB33A0> = <PyQt5.QtWidgets.QComboBox object at 0x00000296F6FB33A0>.findText
E        +      where <PyQt5.QtWidgets.QComboBox object at 0x00000296F6FB33A0> = >> <sas.qtgui.Perspectives.Fitting.UnitTesting.FittingWidgetTest.FittingWidgetMod object at 0x00000296F5C45160>.cbStructureFactor

src\sas\qtgui\Perspectives\Fitting\UnitTesting\FittingWidgetTest.py:251: AssertionError
```

Reply:
I (obviously) can't reproduce that. I noticed in some places that the user's local config files could influence tests - I wonder if that is an issue here. Can you try running the tests with/without a temporary HOME environment variable set?

```HOME=/tmp/pkgs python3 -m pytest --pdb  src/sas/qtgui/Perspectives/Fitting/UnitTesting/FittingWidgetTest.py::FittingWidgetTest::testSelectStructureFactor```

If it fails, `pdb` will launch straight away and the following will provide a list of what is in the structure factor dropdown:

```
(Pdb) for i in range(fittingWindow.cbStructureFactor.count()): print(fittingWindow.cbStructureFactor.itemText(i))
None
hardsphere
hayter_msa
plugin_structure_template
squarewell
stickyhardsphere
```

I think also that we should call fixing the tests to run under Windows as out of scope for _this_ PR, which is only trying to get them to run under Linux. A future PR can enable running the GUI tests under Windows and sort out things like this too — I'll admit to being surprised that there's only one extra test that is failing under Windows.

Reply:
I tested your suggestion and I'm still seeing the same error. The 'plugin_structure_template'  structure factor is not in the list. I agree this should wait.